### PR TITLE
feat: Add spec backwards compatibility

### PR DESCRIPTION
**Pull request by raulb**
Fixes https://github.com/meroxa/turbine-core/issues/71

This pull-request adds backwards compatibility from specs 0.1.1 to 0.2.0. It also:

- Errors if user tries to upgrade from a different spec version of 0.1.1 to 0.2.0
- Validates after building so the build step could be invoked separately while also being validated (needed to be used in Platform API).

### CODE REVIEW

File: pkg/server/record.go [hunk 1]
```diff
@@ -22,7 +22,7 @@ func NewRecordService() *recordService {
 	return &recordService{}
 }
 
-func (s *recordService) Init(ctx context.Context, request *pb.InitRequest) (*emptypb.Empty, error) {
+func (s *recordService) Init(_ context.Context, request *pb.InitRequest) (*emptypb.Empty, error) {
```
Comment (anna-cross) [inline pkg/server/record.go:25]: do we even need to pass context to these functions?

Comment (raulb) [reply pkg/server/record.go:25]: The signature of these functions are autogenerated, and I think it's a good idea we keep them around.

**Review by anna-cross — APPROVED** [review]
All seems good to me here!

**Review by dianadoherty — APPROVED** [review]
Looks amazing

File: pkg/ir/spec.go [hunk 2]
```diff
@@ -201,7 +219,91 @@ func (d *DeploymentSpec) BuildDAG() (*dag.DAG, error) {
...
+}
+
+// upgradeToLatestSpecVersion will ensure that simple topologies as defined in 0.1.1 are still compatible
+// currently, only supported migration is from 0.1.1 to 0.2.0.
+func (d *DeploymentSpec) upgradeToLatestSpecVersion() error {
+	if d.getSpecVersion() == LatestSpecVersion {
+		return nil
+	}
+
+	if (d.getSpecVersion() != "") && (d.getSpecVersion() != SpecVersion_0_1_1) || (LatestSpecVersion != SpecVersion_0_2_0) {
```
[the remark this answers is not in the document]
Comment (raulb) [reply pkg/ir/spec.go:233]: @dianadoherty made this change https://github.com/meroxa/turbine-core/pull/72/commits/9597a82bde06fa1fa0a2fea9965bba7190a02d3b.

**Pull request merged**